File: Crio/crio_programming_interview_problems-master/ClimbingStairs/ClimbingStairs.java
import java.io.*;
import java.util.*;

public class ClimbingStairs {

    public int climbingStairs(int n) {
        // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        if (n == 1) {
            return 1;
        }
        int first = 1;
        int second = 2;
        for (int i = 3; i <= n; i++) {
            int third = first + second;
            first = second;
            second = third;
        }
        return second;
        // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

    }

    public static void main(String[] args) throws IOException {
        Scanner scanner = new Scanner(System.in);
        int numberOfStairs = scanner.nextInt();
        scanner.close();
        int result = new ClimbingStairs().climbingStairs(numberOfStairs);
        System.out.print(result);
    }
}

File: Crio/crio_programming_interview_problems-master/ClimbingStairs/ClimbingStairs.py
def climbingStairs(numberOfStairs):
    # CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    dp = [0]*(numberOfStairs+2)
    dp[0],dp[1]=0,1
    for i in range(numberOfStairs):
        dp[i+2] = dp[i+1]+dp[i]
    return dp[numberOfStairs+1]
    # CRIO_SOLUTION_END_MODULE_L1_PROBLEMS


if __name__ == '__main__':
    numberOfStairs = int(input())
    result = climbingStairs(numberOfStairs)
    print(result)
